### PR TITLE
test(command-build): don't run on forked repos

### PR DESCRIPTION
**Pull request by erezrokah**
Build tests now require a valid auth token which is not available to forked PRs.
Fixes CI failures in https://github.com/netlify/cli/pull/1758

### CODE REVIEW

**Review by ehmicky — APPROVED** [review]

**Comment by eduardoboucas** [comment]
What does this mean for external contributors that submit a PR? Are we able to run the tests "on our side" before merging?

**Comment by erezrokah** [comment]
> What does this mean for external contributors that submit a PR? Are we able to run the tests "on our side" before merging?

We can run those tests locally and they'll run on the default branch post merge.
Contributors _should also_ run those tests locally, but we don't have a good way (yet) to enforce tests that require GitHub secrets.
I'm planning on adding a mock API server as a part of https://github.com/netlify/cli/pull/1489#discussion_r548092573 so maybe we could fallback to a mock API on forks?

**Comment by eduardoboucas** [comment]
> maybe we could fallback to a mock API on forks?

Sounds good. Until we get there, perhaps there's something we can do internally to ensure we don't merge a PR from a forked repo without running the test suite locally, because going forward the PR status will be misleading and there's nothing stopping us from merging a PR that actually breaks the build.

**Comment by erezrokah** [comment]
> Sounds good. Until we get there, perhaps there's something we can do internally to ensure we don't merge a PR from a forked repo without running the test suite locally, because going forward the PR status will be misleading and there's nothing stopping us from merging a PR that actually breaks the build.

Good point. Let's continue this discussion in https://github.com/netlify/cli/issues/1769

**Pull request merged**
